### PR TITLE
add org.opencontainers.image url and source labels to dockerfiles

### PR DESCRIPTION
**Pull request by sankalp-at-gh**
#### Proposed Changes ####

This PR adds the image metadata labels to rke2-runtime image

#### Types of Changes ####

enhancement

#### Verification ####

check rke2-runtime image labels

#### Linked Issues ####

* https://github.com/rancher/rke2/issues/4238

#### User-Facing Change ####
```release-note
```

### CODE REVIEW

**Comment by sankalp-at-gh** [comment]
Hi @brandond and @matttrach, could you please help with review and approval of this PR.
Thanks a lot !

File: Dockerfile.docs [hunk 1]
```diff
@@ -1,4 +1,6 @@
 FROM squidfunk/mkdocs-material
+LABEL org.opencontainers.image.url="https://hub.docker.com/r/rancher/rke2-runtime"
+LABEL org.opencontainers.image.source="https://github.com/rancher/rke2"
```
Comment (brandond) [inline Dockerfile.docs:3]: This Dockerfile is no longer used; we were just using it to build the docs website before it was moved out to a different repo - to labeling it as published as this image is incorrect.

**Review by brandond — CHANGES_REQUESTED** [review]

File: Dockerfile.docs (hunk 1 above)
Comment (sankalp-at-gh) [reply Dockerfile.docs:3]: totally makes sense, thanks for the feedback !

**Review by brandond — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -149,6 +149,8 @@ COPY --from=runtime-collect / /
 
 FROM ubuntu:24.04 AS test
 ARG TARGETARCH
+LABEL org.opencontainers.image.url="https://hub.docker.com/r/rancher/rke2-runtime"
```
Comment (brandond) [inline Dockerfile:152]: You've added these to the `test` target which is not published. I think you probably want to move them to the `runtime` step a couple lines up, which is the step that builds the actual published rke2-runtime linux image.

**Review by brandond — CHANGES_REQUESTED** [review]
one more issue - I didn't notice that the labels were on the wrong step in the main Dockerfile.

File: Dockerfile (hunk 2 above)
Comment (sankalp-at-gh) [reply Dockerfile:152]: Thank you for point it out, it was not very obvious when I first looked at the dockerfile

**Review by brandond — APPROVED** [review]

**Review by vitorsavian — APPROVED** [review]

**Pull request merged**